src/components/table/table.js

```diff
@@ -343,7 +343,7 @@ export default {
                 this.rowClasses(item),
                 {
                   'b-table-has-details': rowShowDetails,
-                  [this.selectedClass]: this.selectedRows[rowIndex]
+                  ['table-'+this.selectedVariant]: this.selectedRows[rowIndex]
```

Comment:
You should check if `this.dark` is true, and if so, switch from `table-{variant}` to `bg-{variant}`

Reply:
done